internal/transport/transport.go

```diff
@@ -296,6 +278,8 @@ type Stream struct {
 	// contentSubtype is the content-subtype for requests.
 	// this must be lowercase or the behavior is undefined.
 	contentSubtype string
+
+	bufferPool mem.BufferPool
```

Comment:
Do we need this field in the `Stream`?

I see that this is being set from three places:
- When creating a stream from a client transport
- When creating a stream from a server transport
- When creating a stream from a server handler transport

`Stream.bufferPool` is accessed from methods on the client transport or server transport, both named `handleData`, and in both places the same buffer pool is available in the transport as well.

Am I missing something?

Reply:
Yeah it was because the transports are defined like this:
```
	st           ServerTransport    // nil for client side Stream
	ct           *http2Client       // nil for server side Stream
```
It works for `ct` since it's a raw type but `st` is an interface which doesn't expose its underlying buffer pool. I did just update the interface to allow this and removed this field